solidity-v1/dashboard/src/components/threshold/AllocationProgressBar.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { useMemo } from "react"
+import ProgressBar, {
+  ProgressBarLegendContext,
+  renderProgressBarLegendItem,
+} from "../ProgressBar"
+import { colors } from "../../constants/colors"
+import OnlyIf from "../OnlyIf"
+import { add, gt } from "../../utils/arithmetics.utils"
+
+const AllocationProgressBar = ({
+  title,
+  currentValue = 0,
+  totalValue = 100,
+  className = "",
+  secondaryValue = null,
+  withLegend = false,
+  currentValueLegendLabel = "",
+  secondaryValueLegendLabel = "",
+  isDataFetching = false,
+}) => {
+  const percentageValue = useMemo(() => {
+    if (isDataFetching) {
+      return "--"
+    }
+
+    if (gt(totalValue, 0)) {
+      return Math.round(
+        (add(currentValue, secondaryValue | 0) / totalValue) * 100
+      )
+    } else {
+      return 0
+    }
+  }, [currentValue, secondaryValue, totalValue, isDataFetching])
+
+  return (
+    <div className={`allocation-progress-bar ${className}`}>
+      <div>
+        <h5 className="text-grey-60">{title}</h5>
+        <div className="allocation-progress-bar__progress-bar-container">
+          <ProgressBar
+            total={isDataFetching ? 100 : totalValue}
+            bgColor={colors.grey20}
+          >
+            <ProgressBar.Inline
+              className="allocation-progress-bar__progress-bar-wrapper"
+              height={20}
+            >
+              <ProgressBar.InlineItem
+                value={
+                  isDataFetching ? 0 : add(currentValue, secondaryValue | 0)
+                }
+                color={colors.secondary}
+              />
+              <OnlyIf condition={!!secondaryValue}>
+                <ProgressBar.InlineItem
+                  value={isDataFetching ? 0 : secondaryValue}
+                  color={colors.yellowSecondary}
+                />
+              </OnlyIf>
+            </ProgressBar.Inline>
+            <OnlyIf condition={withLegend}>
+              <div className="allocation-progress-bar__progress-bar-legend">
+                <ProgressBarLegendContext.Provider
+                  value={{ renderValuePattern: () => {} }}
+                >
+                  {renderProgressBarLegendItem(
+                    {
+                      value: secondaryValue,
+                      label: secondaryValueLegendLabel,
+                      color: colors.yellowSecondary,
+                    },
+                    "legend-item__pending-withdrawal",
+                    "allocation-progress-bar__progress-bar-legend-item"
+                  )}
+                  {renderProgressBarLegendItem(
+                    {
+                      value: currentValue,
+                      label: currentValueLegendLabel,
+                      color: colors.secondary,
+                    },
+                    "legend-item__staked",
+                    "allocation-progress-bar__progress-bar-legend-item"
+                  )}
```

Comment:
Can we use `ProgressBar.LegendItem` component? 

```
<ProgressBar.LegendItem
 value={currentValue.toString()}
 label={currentValueLegendLabel}
 color={colors.secondary}
 className={"legend-item__staked"}
/>
```

Reply:
Fixed in 0f11cc1db404ce9fade9e48fabeb4bc1cc144da3

Reply:
Fixed in 4aa9481a2a117f54f195404001e0628f7a796a2d